fleetspeak/src/server/comms.go

```diff
@@ -302,16 +302,26 @@ func (c commsContext) findMessagesForClient(ctx context.Context, info *comms.Con
 	}
 
 	mids := make([]common.MessageID, 0, len(msgs))
+	var specialMidsToAck []common.MessageID
 	for _, m := range msgs {
 		id, err := common.BytesToMessageID(m.MessageId)
 		if err != nil {
 			return nil, err
 		}
 		mids = append(mids, id)
+		if m.MessageType == "Die" && m.Destination != nil && m.Destination.ServiceName == "system" {
+			specialMidsToAck = append(specialMidsToAck, id)
+		}
```

Comment:
Strictly speaking, if we don't expect an ack for the message, we should also add the corresponding tokens back. Or not subtract them at all - see info.SubtractMessageTokens() above. However, if the message goes through, the client and connection will die soon and this doesn't matter.

I guess maybe add a comment that this is safe only because the connection will be reset if the message goes through? In general, leaking tokens would be a very annoying bug.